### PR TITLE
Add switch frame support and switch-frame-interval CLI option.

### PR DESCRIPTION
**Pull request by tdaede**

### CODE REVIEW

File: src/api/config.rs [hunk 1]
```diff
@@ -64,6 +64,9 @@ pub struct EncoderConfig {
   /// Flag to force all frames to be error resilient.
   pub error_resilient: bool,
 
+  /// Interval between switch frames (0 to disable)
+  pub switch_frame_interval: u64,
+
```
Comment (barrbrain) [inline src/api/config.rs]: We might want to include this in the C-API. See `option_match` in `src/capi.rs`.
Validating the "multiple of `group_input_len`" constraint may not be straightforward.

Comment (tdaede) [reply src/api/config.rs]: It's a pretty terrible constraint in general and I want to come up with a better one.

Comment (lu-zero) [reply src/api/config.rs]: I guess adding a placement override could wait for that :)

Comment (tdaede) [reply src/api/config.rs]: I do want to add a placement override to the API, but again it has to change the pyramid length.

Comment (lu-zero) [reply src/api/config.rs]: Shall we land it as-is or wait for that?

Comment (tdaede) [reply src/api/config.rs]: I made it reject non-low-latency case, so this is worked around for now.

**Comment by tdaede** [comment]
I'm going to land this with it restricted to low latency only if that's OK.

**Comment by lu-zero** [comment]
Two tests fail as in 4 not being a valid interval.

**Comment by tdaede** [comment]
They actually failed because I forgot to delete the reordering test after disallowing reordering with switch frames.

**Pull request merged**